crates/oxc_linter/src/rules/eslint/max_lines.rs

```diff
@@ -0,0 +1,320 @@
+use serde_json::Value;
+
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{CompactStr, Span};
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(max-lines): {0:?}")]
+#[diagnostic(severity(warning), help("Reduce the number of lines in this file"))]
+struct MaxLinesDiagnostic(CompactStr, #[label] Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct MaxLines(Box<MaxLinesConfig>);
+
+#[derive(Debug, Clone)]
+pub struct MaxLinesConfig {
+    max: usize,
+    skip_blank_lines: bool,
+    skip_comments: bool,
+}
+
+impl std::ops::Deref for MaxLines {
+    type Target = MaxLinesConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Default for MaxLinesConfig {
+    fn default() -> Self {
+        Self {
+            max: 300,
+            skip_blank_lines: false,
+            skip_comments: false,
+        }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforce a maximum number of lines per file
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Some people consider large files a code smell. Large files tend to do a lot of things and can make it hard following what’s going.
+    /// While there is not an objective maximum number of lines considered acceptable in a file, most people would agree it should not be in the thousands. Recommendations usually range from 100 to 500 lines.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// ```
+    MaxLines,
+    correctness
```

Comment:
There are several categories in our linter, Clippy has a [similar rule](https://rust-lang.github.io/rust-clippy/master/index.html#/too_many_lines), its category is `pedantic`, what about we keep the same with it?

https://github.com/oxc-project/oxc/blob/32504cadb745c4f81be89fa74fb317822daee1ed/crates/oxc_linter/src/rule.rs#L33-L54

Reply:
Thanks for the hint!